src/reconcile/rec_row.c

```diff
@@ -628,6 +628,41 @@ __rec_row_leaf_insert(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins)
     return (0);
 }
 
+/*
+ * __rec_simplify_time_window --
+ *     Where possible modify time window values to avoid writing them to the cell later.
+ */
+static inline void
+__rec_simplify_time_window(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
+{
+    wt_timestamp_t pinned_ts;
+    uint64_t oldest_id;
+    /*
+     * Get a copy of the pinned timestamp and oldest ID. It is OK if these are out of date, they
+     * only move forward, so the worst case is that we store some information that is no longer
+     * relevant.
+     */
+    __wt_txn_pinned_timestamp(session, &pinned_ts);
+    oldest_id = __wt_txn_oldest_id(session);
+    if (!tw->prepare) {
+        if (tw->stop_txn == WT_TXN_MAX && tw->start_txn < oldest_id) {
+            tw->start_txn = WT_TXN_NONE;
+        }
+        if (tw->durable_stop_ts != WT_TS_NONE && tw->durable_stop_ts < pinned_ts) {
+            tw->durable_stop_ts = WT_TS_NONE;
+        }
+
+        if (tw->stop_ts == WT_TS_MAX && tw->start_ts < pinned_ts) {
+            tw->start_ts = WT_TS_NONE;
+        }
+
+        if (tw->durable_start_ts < pinned_ts) {
+            tw->durable_start_ts = WT_TS_NONE;
+        }
+        WT_ASSERT(session, tw->start_txn <= tw->stop_txn);
+    }
+}
```

Comment:
@luke-pearson, it might be easier to do this work in cell.i, right as we pack the value?

Reply:
Yeah but then our cells won't have their timewindows correctly aggregated as part of the call to: `        __wt_time_aggregate_update(&r->cur_ptr->ta, &tw);` This causes issues in verification and may create hard to find bugs later on.